src/compile/data/optimize.ts

```diff
@@ -161,6 +161,7 @@ export function optimizeDataflow(dataComponent: DataComponent) {
 
   roots.forEach(moveFacetDown);
   roots.forEach(mergeParse);
+  roots.forEach(optimizers.mergeIdenticalTransforms);
```

Comment:
Wouldn't it be better to start from the leaves?

Reply:
Yeah I think so. But I thought it would be better to have the top-down version working with the `mergeParse`(which is also top-down) and refactor them both to bottom-up when I've got the multiple pass thing working.

Reply:
Okay, just don't forget